Backend/TimeTracker.Api/Controllers/ProjectController.cs

```diff
@@ -1,372 +1,429 @@
+        [Authorize]
+        [HttpPost("Tag")]
+        public async Task<GenericResponseDTO<int>> AddTag(CreateTagDTO newTag)
+        {
+            var currentUserId = authHelper.GetCurrentUserId(User);
+
+            // Only allow the teacher to tag a project
+            var project = await database.Projects
+                .AsQueryable()
+                .FirstAsync(x => x.Id == newTag.ProjectId && x.Teacher.Id == currentUserId && x.ArchivedDate == null);
+            
+            if (project == null)
+            {
+                return new GenericResponseDTO<int>()
+                {
+                    Message = "Couldn't find the project",
+                    Success = false
+                };
+            }
+
+            var tag = new Tag()
+            {
+                Name = newTag.Tag,
+                Project = project
+            };
+
+            await database.AddAsync(tag);
+            await database.SaveChangesAsync();
+
+            return new GenericResponseDTO<int>()
+            {
+                Data = tag.Id,
+                Success = true
+            };
+        }
+
+        /// <summary>
+        /// Adds a user to a project
+        /// </summary>
+        /// <param name="projectInvite">The invite code of the project that the user is added to</param>
+        /// <returns>The Id of the project the user was added to</returns>
+        [Authorize]
+        [HttpPost("AddUserToProject")]
+        public async Task<GenericResponseDTO<int>> AddUserToProject(AddUserToProjectDTO inviteCode)
+        {
+            var curUser = await database.Users
+                .Include(x => x.Projects)
+                .FirstOrDefaultAsync(x => x.Id == authHelper.GetCurrentUserId(User));
+
+            Project project = await database.Projects
+                .AsQueryable()
+                .FirstOrDefaultAsync(x => x.InviteCode == inviteCode.InviteCode && x.ArchivedDate == null);
+
+            if (project == null)
+            {
+                return new GenericResponseDTO<int>()
+                {
+                    Message = "Couldn't find the project",
+                    Success = false
+                };
+            }
+
+            curUser.Projects.Add(project);
+            
+            await database.SaveChangesAsync();
+
+            return new GenericResponseDTO<int>()
+            {
+                Data = project.Id,
+                Success = true
+            };
+        }
+
+        /// <summary>
+        /// Removes user from project along with all of their data & entries.
+        /// </summary>
+        /// <param name="projectUserRemoveInfo"> The project id the user is to be removed from, and the user id to be removed </param>
+        /// <returns>The Id of the user deleted</returns>
+        [Authorize]
+        [HttpPatch("RemoveUserFromProject")]
+        public async Task<GenericResponseDTO<int>> RemoveUserFromProject(ProjectRemoveUserDTO projectUserRemoveInfo)

```

Comment:
Could you add a test for this endpoint, we have tests for all the other API calls so might as well keep doing it.